pkg/core/interop/runtime/util.go

```diff
@@ -2,11 +2,50 @@ package runtime
 
 import (
 	"github.com/nspcc-dev/neo-go/pkg/core/interop"
+	"github.com/nspcc-dev/neo-go/pkg/core/state"
+	"github.com/nspcc-dev/neo-go/pkg/util"
 	"github.com/nspcc-dev/neo-go/pkg/vm"
+	"github.com/nspcc-dev/neo-go/pkg/vm/stackitem"
+	"github.com/pkg/errors"
 )
 
 // GasLeft returns remaining amount of GAS.
 func GasLeft(_ *interop.Context, v *vm.VM) error {
 	v.Estack().PushVal(int64(v.GasLimit() - v.GasConsumed()))
 	return nil
 }
+
+// GetNotifications returns notifications emitted by current contract execution.
+func GetNotifications(ic *interop.Context, v *vm.VM) error {
+	item := v.Estack().Pop().Item()
+	notifications := ic.Notifications
+	if _, ok := item.(stackitem.Null); !ok {
+		b, err := item.TryBytes()
+		if err != nil {
+			return err
+		}
+		u, err := util.Uint160DecodeBytesBE(b)
+		if err != nil {
+			return err
+		}
+		notifications = []state.NotificationEvent{}
+		for i := range ic.Notifications {
+			if ic.Notifications[i].ScriptHash.Equals(u) {
+				notifications = append(notifications, ic.Notifications[i])
+			}
+		}
+	}
+	if len(notifications) > vm.MaxStackSize {
+		return errors.New("notification count is too big")
+	}
+	arr := stackitem.NewArray(make([]stackitem.Item, 0, len(notifications)))
+	for i := range notifications {
+		ev := stackitem.NewArray([]stackitem.Item{
+			stackitem.NewByteArray(notifications[i].ScriptHash.BytesBE()),
+			notifications[i].Item,
```

Comment:
Suppose it's an `Array`, now I can change it in the script and affect already emitted notification.

Reply:
I thought about this, but it seems that the original behavior is the same. In C# implementation no copies are made in both `Notify` and `GetNotifications` syscalls.

Reply:
Hmm. Can we try some fancy notify-getnotifications-change-them on testnet? I don't think anyone intended these notifications to be updateable.

Reply:
Try this script `57040012111013c07068115511c04195016f61210b115541274335f121716910ce11ce10ce7200286a1151d06811ce6812ce9e40` .
It is for the following contract:
```golang
package enginecontract

import (
	"github.com/nspcc-dev/neo-go/pkg/interop/runtime"
)

// Main is that famous Main() function, you know.
func Main() int {
	a := []int{0, 1, 2}
	runtime.Notify(a)

	notifications := runtime.GetNotifications(nil)
	notif := notifications[0][1].([]interface{})[0].([]int)
	notif[1] = 40
	return a[1] + a[2]
}
```

Got following response:
```json
{
    "jsonrpc": "2.0",
    "id": 1,
    "result": {
        "script": "57040012111013c07068115511c04195016f61210b115541274335f121716910ce11ce10ce7200286a1151d06811ce6812ce9e40",
        "state": "FAULT",
        "gas_consumed": "2646850",
        "stack": [
            {
                "type": "Integer",
                "value": "42"
            }
        ]
    }
}
```

Maybe it is a bug, but we should be compatible with `preview2`.

Reply:
> Got following response

Argh, it doesn't show notifications for some reason (looks like this field was removed).

> we should be compatible with preview2.

No doubt about that, but I think it's worth submitting a bug for neo.

Reply:
https://github.com/neo-project/neo/issues/1707